pkg/apimachinery/discoverysummarizer/types.go

```diff
@@ -0,0 +1,47 @@
+/*
+Copyright 2016 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package discoverysummarizer
+
+type FederatedServerList struct {
+	Servers []FederatedServer `json:"servers"`
+}
+
+type FederatedServer struct {
+	ServerIP string `json:"serverIP"`
+}
+
+type APIGroup struct {
+	// name is the name of the group.
+	Name string `json:"name"`
+	// The IP address of the server that is serving this group.
+	// TODO: Support returning both internal and external IP addresses:
+	// http://issues.k8s.io/20229
+	ServerIP string `json:"serverIP"`
```

Comment:
Why are we assuming this is an API address.  This should be a HostName that can also be an IP.